### PR TITLE
Add gateway configuration and standalone gateway script

### LINKED ISSUE #1524: The gateway can be configured
I can configure the gateway using a toml file. It's the same for embedded and standalone scenario. It is separate from the broker config file. There also should be a bash/bat file to start the standalone gateway. The broker config contains a flag which defines if the gateway should be started embedded. Most of the configuration parameters can be set as environment variable.

### LINKED ISSUE #1532: I can use zbctl inside the docker image without specifing the broker address
Currently I have to specify the address of the broker to run zbctl inside the docker container like this

```
$ docker exec zeebe bash -c 'ZB_BROKER_ADDR=$(hostname -i):26500 zbctl status'
```

The reason is that on startup the broker binds to the external interface of the Container to get the topology announcement right to use it with an external gateway. 

Possible solution is to bind the embedded gateway to the local or all interfaces. The zbctl inside the container should work out of the box.

### PR DESCRIPTION
**Pull request by menski**
- add configuration classes to gateway module
- add embedded gateway configuration to broker configuration
- add script and configuration to start standalone gateway in distribution

closes #1524
closes #1532

### CODE REVIEW

File: dist/src/main/config/zeebe.cfg.toml [hunk 1]
```diff
@@ -30,6 +30,41 @@
...
+
+[gateway]
+# Enable the embedded gateway to start on broker startup
+#
+# This setting can also be overridden using the environment variable ZEEBE_EMBED_GATEWAY.
+# enable = true
+
+[gateway.network]
+# Sets the host the embedded gateway binds to
+#
```
Comment (Zelldon) [inline dist/src/main/config/zeebe.cfg.toml]: Maybe for the user is also interested in the order of precedence. What do you think?

File: dist/src/main/config/zeebe.cfg.toml [hunk 2]
```diff
@@ -30,6 +30,41 @@
 
 # ----------------------------------------------------
 
+
+[gateway]
+# Enable the embedded gateway to start on broker startup
+#
```
Comment (Zelldon) [inline dist/src/main/config/zeebe.cfg.toml]: Can we remove some of the blank lines ?

File: dist/src/main/config/zeebe.cfg.toml (hunk 1 above)
Comment (Zelldon) [reply dist/src/main/config/zeebe.cfg.toml]: I mean this:
```
    // network host precedence from higher to lower:
    // 1. ENV_GATEWAY_HOST
    // 2. specified in gateway toml section
    // 3. ENV_HOST
    // 4. specified in broker network toml section
```

File: dist/src/main/config/zeebe.cfg.toml [hunk 3]
```diff
@@ -30,6 +30,41 @@
...
+# port = 26500
+
+[gateway.cluster]
+# Sets the broker the gateway should initial contact
+#
+# This setting can also be overridden using the environment variable ZEEBE_GATEWAY_CONTACT_POINT.
+# contactPoint = "127.0.0.1:26501"
+
+# Sets size of the transport buffer to send and received messages between gateway and broker cluster
+#
```
Comment (Zelldon) [inline dist/src/main/config/zeebe.cfg.toml]: How large the transport buffer should be? Maybe we can explain in the config our properties a bit more in detail what this means. What do you think?

**Review by ChrisKujawa — APPROVED** [review]
bors r+

**Pull request merged**